drivers/dma/dma_si32.c

```diff
@@ -54,6 +54,8 @@ static struct dma_si32_data dma_si32_data = {.ctx = {
 
 __aligned(SI32_DMADESC_PRI_ALIGN) struct SI32_DMADESC_A_Struct channel_descriptors[CHANNEL_COUNT];
 
+static int dma_si32_stop(const struct device *dev, const uint32_t channel);
```

Comment:
How about just moving the definition of `dma_si32_stop()` before `dma_si32_config()` and then you wouldn't need the forward declaration?

Reply:
Did not do because I wanted to avoid noise. But you are right, better now 1x some Git log noise, but making it easier to maintain the code in the future.

Reply:
You can always put the move into a separate "preparatory" commit before the one that does the actual changes. That way it's still easy to see what the actual functional changes are and there's only "noise" when looking at the diff for the entire PR.

Reply:
done